code/game/turfs/turf.dm

```diff
@@ -522,9 +522,13 @@
 			return I
 	return I
 
-/turf/AllowDrop()
+/turf/AllowDrop()// turf always valid place to drop something
 	return TRUE
 
+
+/turf/drop_location()// turf always valid place to drop something
+	return src
```

Comment:
With this drop_location() dont drop to area i think.
dont know why but with no this
```
/mob/dead/forceMove(atom/destination)
	var/turf/old_turf = get_turf(src)
	var/turf/new_turf = destination.drop_location()
	if (old_turf?.z != new_turf?.z)
		onTransitZ(old_turf?.z, new_turf?.z)
	var/oldloc = loc
	loc = destination
	Moved(oldloc, NONE, TRUE)
```
call area/drop_location()

Reply:
`var/turf/new_turf = destination.drop_location()` is a problem by itself, if you're looking for a turf just keep get_turf

Reply:
you can keep the drop location code in turf to catch potential mistakes, but generally speaking drop_location should never end up being called on a turf directly

Reply:
roundstart initialization push turf to here.
landmarks save loc to global list and delete self
then init push picked lok to here

Reply:
i'm not exactly sure about landmark code, but generally speaking if you _always_ expect a turf as a result, you should use get_turf() for code readability

Reply:
Yes i see it clearly now. It use it only to check Z